### DISCUSSION TITLE
GPS data not stored in .pcapng file

### DISCUSSION THREAD

**Discussion by D4zz4** [ZerBea/hcxdumptool#255] [Q&A]
Hey everyone. I need some help with hcxdumptool.
I am trying to store gps data along with the handshakes to later visualize it.
This is the command I am using:
hcxdumptool -i wlan1 -o buttondumpgpsd.pcapng --disable_client_attacks --disable_ap_attacks --enable_status=35 --use_gpsd

While running, it shows the GPS data every 60 seconds, but I can't find it in the .pcapng file after exiting the process.
Did I miss something here? Thank you all for your time and help
![Dfm7YcIL5c](https://user-images.githubusercontent.com/76969297/223185816-1f026081-f11f-4710-b0d9-c0a21457e57b.png)

**Comment by ZerBea** [comment]
hcxdumptool use pcpng comment fields in which this (and several other information e.g. NONCE, hashes, REPLAYCOUNT,..) is stored.
hcxpcapngtool evaluate this fields directly from the pcapng dump file and convert the data on request by option switch:

from hcxpcapngtool --help:
```
--nmea=<file>                      : output GPS data in NMEA 0183 format
                                     format: NMEA 0183 $GPGGA, $GPRMC, $GPWPL
                                     to convert it to gpx, use GPSBabel:
                                     gpsbabel -i nmea -f hcxdumptool.nmea -o gpx,gpxver=1.1 -F hcxdumptool.gpx
                                     to display the track, open file.gpx with viking
--csv=<file>                       : output ACCESS POINT information in CSV format
                                     delimiter: tabulator (0x08)
                                     columns:
                                     YYYY-MM-DD HH:MM:SS MAC_AP ESSID ENC_TYPE CIPHER AKM COUNTRY_INFO CHANNEL RSSI GPS(DM.m) GPS(D.d) GPSFIX SATCOUNT HDOP ALTITUDE UNIT
                                     to convert it to other formats, use bash tools or scripting languages
                                     GPS FIX:
                                     0 = fix not available or invalid
                                     1 = fix valid (GPS SPS mode)
                                     2 = fix valid (differential GPS SPS Mode)
                                     3 = not supported
                                     4 = not supported
                                     5 = not supported
                                     6 = fix valid (Dead Reckoning Mode)
```

BTW:
hcxpcapngtool store its information in the MESSAGEPAIR field to give hashcat an information how to handle the hash.
The entire information chain:
`hcxdumptool (store additional information in pcapng comment fields) -> hcxpcapngtool (store additional information in MESSAGEPAIR field) -> hashcat
`

You could also try hcxlabtool which also has a GPS option (not via GPSD - instead it read the data directly from the device). Otuput is stored using NMEA 0183 format.

I decided to use NMEA 0183, because it is a standard with nearly no(!) overhead in its messages.
https://en.wikipedia.org/wiki/NMEA_0183

**Reply by D4zz4** [reply]
First of all, thank you very much for your reply, and sorry for my late response.
I found the GPS data in pcapng file, but I saw that it had the exact same coordinates in every package.
I know for sure that the GPS Dongle (U-Blox7) was getting valid coordinates during my last wardriving session because I could see it on gpsmon. I am not sure why it does not update the coordinates in the capture file though.
Is there something wrong with my command?
I will try hcxlabtool the next few days and will report if the results are better.
Thanks for your time

**Comment by ZerBea** [comment]
Does timestamp and koorinates changes in status output (every second) or are they always the same.

How is the output of hcxpcapngtool?
`$ hcxpcapngtool --nmea=gps.nmea --csv=info.csv
`
Are timestamps and coordinates different or are they all the same?


If you add nmea to the command line, gps data are written directly to a file. Do the coordinates ant the timestamp change?
`--nmea=<file>                      : save track to file
`

To make sure it is not a broken socket to GPSD, please run hcxdumptool with option "use_gps_device=". Do not run GPSD.
How about gps data. Do they change, now?

It would be helpful if you add an example pcapng file. 10 minutes record should be enough.

**Comment by ZerBea** [comment]
If you don't want to add the pcapng file here, you can send it via email.
You can get my email address via git API:
https://api.github.com/users/ZerBea/events/public

**Comment by ZerBea** [comment]
By latest commit:
https://github.com/ZerBea/hcxdumptool/commit/06d3c9def89e8ca8d8a17c27337c37ff652507fd
I added a watchdog to hcxdumptool. It should print a warning if the connection to GPSD socket is broken.
